transport/tchannel/config_test.go

```diff
@@ -0,0 +1,242 @@
+			cfg:        attrs{"tchannel": attrs{}},
+			wantErrors: []string{"inbound address is required"},
+		},
+		{
+			desc: "too many inbounds",
+			cfg: attrs{
+				"tchannel":  attrs{"address": ":4040"},
+				"tchannel2": attrs{"address": ":4041", "type": "tchannel"},
+			},
+			wantErrors: []string{"at most one TChannel inbound may be specified"},
+		},
+		{
+			desc:       "WithChannel fails",
+			cfg:        attrs{"tchannel": attrs{"address": ":4040"}},
+			opts:       []Option{WithChannel(someChannel)},
+			wantErrors: []string{"TransportSpec does not accept WithChannel"},
+		},
+		{
+			desc:       "ServiceName fails",
+			cfg:        attrs{"tchannel": attrs{"address": ":4040"}},
+			opts:       []Option{ServiceName("zzzzzzzzz")},
+			wantErrors: []string{"TransportSpec does not accept ServiceName"},
+		},
+		{
+			desc:       "ListenAddr fails",
+			cfg:        attrs{"tchannel": attrs{"address": ":4040"}},
+			opts:       []Option{ListenAddr(":8080")},
+			wantErrors: []string{"TransportSpec does not accept ListenAddr"},
+		},
+	}
+
+	outboundTests := []outboundTest{
+		{desc: "no outbound", empty: true},
+		{
+			desc: "simple outbound",
+			cfg: attrs{
+				"myservice": attrs{
+					"tchannel": attrs{
+						"peer": "127.0.0.1:4040",
+					},
+				},
+			},
+		},
+		{
+			desc: "outbound interpolation",
+			env:  map[string]string{"SERVICE_PORT": "4040"},
+			cfg: attrs{
+				"myservice": attrs{
+					"tchannel": attrs{
+						"peer": "127.0.0.1:${SERVICE_PORT}",
+					},
+				},
+			},
+		},
+		{
+			desc: "outbound bad peer list",
+			cfg: attrs{
+				"myservice": attrs{
+					"tchannel": attrs{"least-pending": "wat"},
+				},
+			},
+			wantErrors: []string{
+				`failed to configure unary outbound for "myservice"`,
+				`failed to read attribute "least-pending": wat`,
+			},
+		},
+	}
+
+	runTest := func(t *testing.T, inbound inboundTest, outbound outboundTest) {
+		env := make(map[string]string)
+		for k, v := range inbound.env {
+			env[k] = v
+		}
+		for k, v := range outbound.env {
+			env[k] = v
+		}
+		configurator := config.New(config.InterpolationResolver(mapResolver(env)))
+
+		opts := append(inbound.opts, outbound.opts...)
+		err := configurator.RegisterTransport(TransportSpec(opts...))
+		require.NoError(t, err, "failed to register transport spec")
+
+		cfgData := make(attrs)
+		if inbound.cfg != nil {
+			cfgData["inbounds"] = inbound.cfg
+		}
+		if outbound.cfg != nil {
+			cfgData["outbounds"] = outbound.cfg
+		}
+		cfg, err := configurator.LoadConfig("foo", cfgData)
+
+		if len(inbound.wantErrors) > 0 {
+			require.Error(t, err, "expected failure")
+			for _, msg := range inbound.wantErrors {
+				assert.Contains(t, err.Error(), msg)
+			}
+			return
+		}
+
+		if len(outbound.wantErrors) > 0 {
+			require.Error(t, err, "expected failure")
+			for _, msg := range outbound.wantErrors {
+				assert.Contains(t, err.Error(), msg)
+			}
+			return
+		}
+
+		require.NoError(t, err, "expected success")
```

Comment:
Since this test has many stages, might be helpful to make the message a bit longer - "expected success loading config." Likewise with the "expected failure" messages in the two preceding blocks.